### PR TITLE
feat: SAR-Identified Risks

### LINKED ISSUE #656: SAR: Identified Risks- 4.6
**Extended Description**
- As an ASAP Developer, in order to provide FedRAMP reviewers easy confirmation that Identified Risks have been described., I want to provide validations in accordance with section 4.6 of the SAR Guide.

**Acceptance Criteria**
- [x] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [x] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [x] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [x] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- [x] a finding with a target/status/@state of 'not-satisfied' must have at least one associated-risk child
- [x] Any associated-risk must have a @risk-uuid value that points to a risk/@uuid value in the SAR.
- [x] Each risk assembly must have an characterization with children facet elements of @system='https://fedramp.gov' AND @name='likelihood' AND @system='https://fedramp.gov' and @name='impact').
- [x] each facet (likelihood' or 'impact' and 'https://fedramp.gov') have a @value of 'low', 'moderate', or 'high'

**Definition of Done**
 - [x] Acceptance criteria met
 - [x] Unit test coverage of our code > 95%
 - [x] Automated code quality checks passed
 - [x] Security reviewed and reported
 - [x] Reviewed against plain language guidelines
 - [x] Code must be self-documenting
 - [x] No local tech debt
 - [x] Load/performance tests passed – needs to be created/automated
 - [x] Documentation updated
 - [x] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
Close #656 
* A finding with a target/status/@State of 'not-satisfied' must have at least one associated-risk child
*Any associated-risk must have a @risk-uuid value that points to a risk/@uuid value in the SAR.
* Each risk assembly must have an characterization with children facet elements of @System='https://fedramp.gov' AND  @name='likelihood' AND @System='https://fedramp.gov' and @name='impact').
* Each facet (likelihood' or 'impact' and 'https://fedramp.gov') have a @value of 'low', 'moderate', or 'high'

### CODE REVIEW

File: src/validations/rules/sar.sch [hunk 1]
```diff
@@ -176,6 +176,77 @@
...
+                diagnostics="has-finding-target-status-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.5"
+                id="has-finding-target-status-issue"
+                role="error"
+                test="
+                    if (oscal:target/oscal:status/@state = 'not-satisfied')
+                    then
+                        exists(oscal:associated-risk)
+                    else
+                        true()">A finding with a target/status of 'not-satisfied has an associated-risk element.</sch:assert>
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:193]:
```suggestion
                        true()">A finding with a target/status of 'not-satisfied must have an associated-risk element.</sch:assert>
```

File: src/validations/rules/sar.sch [hunk 2]
```diff
@@ -176,6 +176,77 @@
...
+            context="oscal:associated-risk">
+            <sch:let
+                name="SAR-risk-uuids"
+                value="/oscal:assessment-results/oscal:result/oscal:risk/@uuid" />
+            <sch:assert
+                diagnostics="has-associated-risk-matching-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.6"
+                id="has-associated-risk-matching"
+                role="error"
+                test="@risk-uuid[. = $SAR-risk-uuids]">An associated-risk/@risk-uuid has a matching risk/@uuid in the SAR.</sch:assert>
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:206]:
```suggestion
                test="@risk-uuid[. = $SAR-risk-uuids]">An associated-risk/@risk-uuid must have a matching risk/@uuid in the SAR.</sch:assert>
```

File: src/validations/rules/sar.sch [hunk 3]
```diff
@@ -176,6 +176,77 @@
...
+        <!-- It is not clear from the current SAR Guide how to handle multiple facet elements with differing prop[@name='state']/@value values. -->
+        <!--<sch:rule
+            context="oscal:characterization">
+            <sch:assert
+                diagnostics="has-likelihood-and-impact-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.6"
+                id="has-likelihood-and-impact"
+                role="error"
+                see="https://github.com/GSA/fedramp-automation-guides/issues/44"
+                test="">Each characterization has pairs of facet elements where the @name is 'likelihood' or 'impact' and the child prop[@name='state'] have the same @value values.</sch:assert>
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:218]:
```suggestion
                test="">Each characterization must have pairs of facet elements where the @name is 'likelihood' or 'impact' and the child prop[@name='state'] have the same @value values.</sch:assert>
```

File: src/validations/rules/sar.sch [hunk 4]
```diff
@@ -176,6 +176,77 @@
...
+
+        <sch:rule
+            context="oscal:characterization">
+            <sch:assert
+                diagnostics="has-facet-likelihood-and-impact-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.6"
+                id="has-facet-likelihood-and-impact"
+                role="error"
+                test="oscal:facet[@name = 'likelihood' and @system = 'https://fedramp.gov'] and oscal:facet[@name = 'impact' and @system = 'https://fedramp.gov']">Facets
+                with @name of 'likelihood' and @name='impact' exist in the characterization.</sch:assert>
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:229]:
```suggestion
                with @name of 'likelihood' and @name='impact' must exist in the characterization.</sch:assert>
```

File: src/validations/rules/sar.sch [hunk 5]
```diff
@@ -176,6 +176,77 @@
...
+                diagnostics="has-facet-correct-values-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.6"
+                id="has-facet-correct-values"
+                role="error"
+                test="
+                    if ((@name = 'likelihood') and @system = 'https://fedramp.gov' or (@name = 'impact') and @system = 'https://fedramp.gov')
+                    then
+                        @value = ('low', 'moderate', 'high')
+                    else
+                        true()">facet with either @name = 'likelihood' or @name='impact' has an @value of either 'low', 'moderate', or
```
Comment (danielnaab) [inline src/validations/rules/sar.sch:244]:
```suggestion
                        true()">A facet with @name = 'likelihood' or @name='impact' must have an @value of either 'low', 'moderate', or
```

**Review by danielnaab — CHANGES_REQUESTED** [review]

**Review by GaryGapinski — APPROVED** [review]
SAR documentation does not indicate `impact` is required (The POA&M guide does).

**Review by danielnaab — APPROVED** [review]

**Pull request merged**